web/packages/teleterm/src/ui/TopBar/AdditionalActions.tsx

```diff
@@ -36,7 +36,10 @@ type MenuItem = {
   onNavigate: () => void;
   prependSeparator?: boolean;
   keyboardShortcutAction?: KeyboardShortcutAction;
-};
+} & (MenuItemAlwaysEnabled | MenuItemConditionallyDisabled);
+
+type MenuItemAlwaysEnabled = { isDisabled?: false };
+type MenuItemConditionallyDisabled = { isDisabled: true; disabledText: string };
```

Comment:
I'm trying out something new here. I defined these two explicitly as separate types so that the error message is nicer.

What I wanted to achieve is to force any item that has `isDisabled` set to `true` to also specify `disabledText`.